WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java

```diff
@@ -180,11 +180,12 @@ public String getLabel() {
     }
 
     @Override
-    public boolean equals(Object tag){
-        if (tag == null) return false;
-
-        if (!ReaderTag.class.isAssignableFrom(tag.getClass())) return false;
-
-        return getLabel().equals(((ReaderTag) tag).getLabel());
+    public boolean equals(Object object){
+        if (object instanceof ReaderTag) {
+            ReaderTag tag = (ReaderTag) object;
+            return (tag.tagType == this.tagType && tag.getLabel().equals(this.getLabel()));
```

Comment:
I don't think a comment is necessary since it's easy enough to determine that `getLabel()` will never return null. Plus, the `getters` in all of our models I've been involved with should never return null for string values - that was something I made sure of not long after I started.